### ISSUE TITLE
Support optional chaining operator

### ISSUE THREAD

**Issue by devhammed** [lebab/lebab#311]
Great job here.

I want to suggest adding support for the optional chaining operator e.g

Code:

```js
'use strict';

function greet(name) {
  name = name  || 'World'
  var isJohn = name && name.includes && name.includes('John')
  alert(isJohn ? 'Hello Admin' : 'Hello ' + name)
}
```

Current Output:

```js
function greet(name = 'World') {
  const isJohn = name && name.includes && name.includes('John')
  alert(isJohn ? 'Hello Admin' : `Hello ${name}`)
}
```

Expected Output:
```js
function greet(name = 'World') {
  const isJohn = name?.includes('John')
  alert(isJohn ? 'Hi Admin' : `Hello ${name}`)
}
```

**Comment by uniibu** [comment]
Good suggestion, however, even tho the proposal is already on Stage 4(finished), we would still need to wait around July where the schedule(if schedules are followed) of the new standard by the ECMA General Assembly will get approved.

**Comment by frank-dspeed** [comment]
@uniibu thats not correct you can assume every stage 3 feature will get implemented even browsers have implemented it already stage 3 is save to impl

**Comment by uniibu** [comment]
@frank-dspeed thanks for clarifying, although I never said that a stage 3 proposals have a chance to not be implemented.  In fact, I don't think there has ever been a stage 3 proposal that was eventually withdrawn/rejected. You are also correct that browsers indeed are implementing most of the stage 3 proposals, however, Lebab does not rely on browsers as its parser. It uses [Espree](https://github.com/eslint/espree) and during the time i wrote that comment, `optional chaining` was still not implemented.

Going back, I would say this is now possible to do since Espree have already implemented it on their latest release [v7.2.0](https://github.com/eslint/espree/releases/tag/v7.2.0) which includes [PR #446](https://github.com/eslint/espree/pull/446) and was only released 6 days ago.

**Comment by frank-dspeed** [comment]
@uniibu espree now uses acron anyway now under the hood so it supports everything

**Comment by frank-dspeed** [comment]
@uniibu even top level await

**Comment by uniibu** [comment]
For me, I have nothing against adding new features and support latest proposals. But we have to keep in mind that Lebab wasn't made only for codes that runs on browsers in general. For example, a implementing a `top level await` must be put on a separate transform option to be able to disable it as it does not work on nodej environment.

Fwiw, I do not decide which new feature to add or edit for this library. @nene would be the right person for that.

**Comment by frank-dspeed** [comment]
@uniibu we have TOP LEVEL AWAIT inside nodejs :)

**Comment by frank-dspeed** [comment]
@uniibu https://github.com/nodejs/node/pull/30370#issuecomment-628145201

**Comment by uniibu** [comment]
TLA on nodejs is still under `experimental`. I don't see it being implemented fully to nodejs at stage 3 (https://github.com/tc39/proposal-top-level-await), maybe when v15 LTS comes out on October, or maybe not. Besides, to make it work on nodejs at this time, a flag `--harmony-top-level-await` must be used along with `"type": "module"`. From a production stand point, I don't see my self using it until it comes out of `experimental` stage.

On another point, I think the project is dead or is dying/forgotten. @nene has been inactive for a long time now and I don't think owner really cares much at this point. My PR which updates the dependencies including `espree`, has not seen the light of day for 2 months already. It's sad imo, the project had its potential.

**Comment by frank-dspeed** [comment]
@uniibu i will fork lebab now and turn it into codemods

**Comment by DiegoRBaquero** [comment]
Optional chaining works in Node 14+, would be a great addition to lebab
